include/singa/core/tensor.h

```diff
@@ -59,8 +59,11 @@ class Tensor {
   explicit Tensor(Shape &&shape, DataType dtype = kFloat32);
   explicit Tensor(const Shape &shape, DataType dtype = kFloat32);
   Tensor(Shape &&shape, std::shared_ptr<Device> dev, DataType dtype = kFloat32);
+  Tensor(Shape &&shape, vector<int> strides, std::shared_ptr<Device> dev, DataType dtype = kFloat32;
```

Comment:
by default the strides are computed based on the shape.
e.g. stride[0]=size/shape[0], stride[1]=size/shape[0]/shape[1], ... stride[-1]=1;
hence, we do not need the users to specify strides in the constructor.